### DISCUSSION TITLE
face_recognition installation failing on Windows 11

### DISCUSSION THREAD

**Discussion by Ishino** [WASasquatch/was-node-suite-comfyui#68] [General]
Just a quick note for those having trouble when `pip install face_recognition` runs.

ModuleNotFoundError: No module named 'cmake' error on dlib installation.

The way I had to resolve this was to:

- install Visual Studio C++ ( https://visualstudio.microsoft.com/vs/features/cplusplus/ )
- install CMake ( https://cmake.org/download/ )
- download and manual install dlib ( https://pypi.org/project/dlib/#files ) - In the dlib folder, run 'python setup.py install'

After this, installing face_recognition should run just fine.

**Comment by WASasquatch** [comment]
Weird process. Wonder why, I have none of those things on my win11 (besides dlib that it makes) and was able to install.

**Reply by Ishino** [reply]
I found several people who had similar issues with this library. Cmake installation working fine, it's available in PATH, works fine in the shell, but a 'pip install dlib' always fails saying it can't find CMake. Only the drastic manual install worked in the end. I do assume it's environment specific, hence why I opened this as a discussion and not an issue.

**Reply by WASasquatch** [reply]
Curious, did you run your requirements install through the python embedded's python, or system?

This is the downside to the custom_nodes is running `pip install requirements.exe` will install to the system, not ComfyUI. 

Gotta run from the embeds python `C:\ComfyUI_windows_portable\python_embeded\python.exe -m pip install C:\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\requirements.txt`

**Reply by Ishino** [reply]
I'm not using the portable build, installed manually, so yes, system install ( in a venv ).

**Comment by WASasquatch** [comment]
I also may be thinking of replacing face_recognition, since for me at least, it never finds a face, with this: https://github.com/iitzco/faced

What do you think? It seems like a better option because it can run on CPU or GPU it seems, or maybe that's a comparison and it's only CPU.
